h2o-py/h2o/frame.py

```diff
@@ -1280,34 +1284,63 @@ def drop(self, index, axis=1):
           H2OFrame with the respective dropped columns or rows. Returns a new H2OFrame.
         """
         if axis == 1:
-            if is_type(index, str):
-                index = self.names.index(index)
+            if not isinstance(index,list):
+                #If input is a string, i.e., "C1":
+                if is_type(index, str):
+                    #Check if index is an actual column(s) in the frame
+                    assert set([index]).issubset(self.names) == True, "Column(s) selected to drop are not in " \
+                                                                          "original frame: %r" % index
+                    index = self.names.index(index)
+                #If input is an int indicating a column index, i.e., 3:
+                elif is_type(index, int):
+                    #Check if index is an actual column index in the frame
+                    assert index < self.ncol, "Column index selected to drop is not part of the frame: %r" % index
+                    assert index >= 0, "Column index selected to drop is not positive: %r" % index
```

Comment:
Do we really need negative indices? Currently you just need to give the column index or set of column indexes to drop. I feel thats pretty straight forward as is.